core/src/blocktree.rs

```diff
@@ -429,10 +576,68 @@ impl Blocktree {
     pub fn get_data_blob_bytes(&self, slot: u64, index: u64) -> Result<Option<Vec<u8>>> {
         self.data_cf.get_by_slot_index(slot, index)
     }
+
+    #[cfg(not(feature = "erasure"))]
     pub fn put_coding_blob_bytes(&self, slot: u64, index: u64, bytes: &[u8]) -> Result<()> {
         self.erasure_cf.put_by_slot_index(slot, index, bytes)
     }
 
+    #[cfg(feature = "erasure")]
+    pub fn put_coding_blob_bytes(&self, slot: u64, index: u64, bytes: &[u8]) -> Result<()> {
+        let mut erasure_meta = self
+            .erasure_meta_cf
+            .get_by_slot(slot)?
+            .unwrap_or(ErasureMeta {
+                slot,
+                counts: vec![],
+            });
+        let set_index = index as usize / erasure::NUM_DATA;
+        let coding_index = set_index + (index as usize - (erasure::NUM_DATA - erasure::NUM_CODING));
+
+        while erasure_meta.counts.len() < set_index {
+            erasure_meta.counts.push((0, 0));
+        }
+
+        erasure_meta.counts[set_index].1 |= 1 << coding_index;
+
+        let mut writebatch = self.db.batch()?;
+
+        writebatch.put_cf(
+            self.erasure_cf.handle(),
+            &ErasureCf::key(slot, index),
+            bytes,
+        )?;
+        writebatch.put_cf(
+            self.erasure_meta_cf.handle(),
+            &ErasureMetaCf::key(slot),
+            &serialize(&erasure_meta)?,
+        )?;
+
+        self.db.write(writebatch)?;
+
+        if erasure_meta.can_recover(set_index as u64) {
```

Comment:
Is this can_recover check needed on insertion of data blobs as well?

Reply:
Yes because whether recovery can be done depends on how many data blobs are missing, so adding one could enable the recovery for 2-3 more if there were already 11/16 nd 4/4 coding blobs (I think, will check with sakridge to make sure)